### PR TITLE
Allow passing a custom configuration object to the request module

### PR DESCRIPTION
**Pull request by ExtraHash**

### CODE REVIEW

File: lib/Daemon.ts [hunk 1]
```diff
@@ -548,7 +554,7 @@ export class Daemon extends EventEmitter implements IDaemon {
             );
 
             const data = await request({
-                ...options,
+                ...requestOptions,
```
Comment (zpalmtree) [inline lib/Daemon.ts]: Why not just
```javascript
const data = await request({
    ...options,
    ...this.config.customRequestOptions
});
```

Comment (ExtraHash) [reply lib/Daemon.ts]: I guess I wasn't aware of the `...` syntax, that concats two objects or something?

Comment (zpalmtree) [reply lib/Daemon.ts]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

File: lib/Daemon.ts [hunk 2]
```diff
@@ -530,6 +530,7 @@ export class Daemon extends EventEmitter implements IDaemon {
      * Makes a get request to the given endpoint
      */
     private async makeRequest(endpoint: string, method: string, body?: any): Promise<any> {
+
```
Comment (zpalmtree) [inline lib/Daemon.ts]:
```suggestion
```

**Comment by zpalmtree** [comment]
Thanks for the PR!

**Pull request merged**